geode-core/src/test/java/org/apache/geode/internal/JarDeployerIntegrationTest.java

```diff
@@ -126,14 +111,16 @@ public void testDeployToInvalidDirectory() throws IOException, ClassNotFoundExce
       public void run() {
         try {
           barrier.await();
-        } catch (Exception e) {
-          fail(e);
+        } catch (InterruptedException iex) {
```

Comment:
Never catch "unexpected" exceptions in a JUnit Test. This makes a mess and adds tons of lines to a test for no good reason. JUnit framework handles "unexpected" exceptions than this crazy catch/rethrow business. All the books about JUnit say don't do this. Also, use new Concurrency constructs.
```java
    Future<Boolean> done = Executors.newSingleThreadExecutor().submit(() -> {
      barrier.await();
      jarDeployer.deployWithoutRegistering("JarDeployerIntegrationTest.jar", jarBytes);
      return true;
    });

    barrier.await();
    Thread.sleep(500); // don't ever do this
    alternateDir.mkdir();
    assertThat(done.get(2, TimeUnit.MINUTES)).isTrue();
```